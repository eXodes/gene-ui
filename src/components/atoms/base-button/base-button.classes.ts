export const classNames = {
  appearance: {
    fill: {
      default:
        "g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600",
      primary:
        "g-bg-primary-600 g-text-clear hover:g-bg-primary-500 focus:g-ring-primary-600",
      attention:
        "g-bg-attention-600 g-text-clear hover:g-bg-attention-500 focus:g-ring-attention-600",
      success:
        "g-bg-success-600 g-text-clear hover:g-bg-success-500 focus:g-ring-success-600",
      warning:
        "g-bg-warning-600 g-text-clear hover:g-bg-warning-500 focus:g-ring-warning-600",
      danger:
        "g-bg-danger-600 g-text-clear hover:g-bg-danger-500 focus:g-ring-danger-600",
      disabled: "g-bg-gray-200 g-text-gray-400 g-cursor-not-allowed",
    },
    clear: {
      default:
        "g-bg-transparent g-text-gray-600 hover:g-bg-gray-100 focus:g-ring-gray-600 dark:g-text-gray-100 dark:hover:g-bg-gray-800",
      primary:
        "g-bg-transparent g-text-primary-600 hover:g-bg-primary-100 focus:g-ring-primary-600 dark:hover:g-bg-primary-800 dark:hover:g-bg-opacity-25",
      attention:
        "g-bg-transparent g-text-attention-600 hover:g-bg-attention-100 focus:g-ring-attention-600 dark:hover:g-bg-attention-800 dark:hover:g-bg-opacity-25",
      success:
        "g-bg-transparent g-text-success-600 hover:g-bg-success-100 focus:g-ring-success-600 dark:hover:g-bg-success-800 dark:hover:g-bg-opacity-25",
      warning:
        "g-bg-transparent g-text-warning-600 hover:g-bg-warning-100 focus:g-ring-warning-600 dark:hover:g-bg-warning-800 dark:hover:g-bg-opacity-25",
      danger:
        "g-bg-transparent g-text-danger-600 hover:g-bg-danger-100 focus:g-ring-danger-600 dark:hover:g-bg-danger-800 dark:hover:g-bg-opacity-25",
      disabled: "g-bg-transparent g-text-gray-400 g-cursor-not-allowed",
    },
  },
  size: {
    small: "g-py-1 g-px-2 g-text-xs focus:g-ring-2",
    medium: "g-py-1.5 g-px-3 g-text-sm focus:g-ring-4",
    large: "g-py-2 g-px-4 g-text-md focus:g-ring-4",
  },
};
