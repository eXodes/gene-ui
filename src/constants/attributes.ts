export enum APPEARANCE {
  DEFAULT = "default",
  PRIMARY = "primary",
  ATTENTION = "attention",
  SUCCESS = "success",
  WARNING = "warning",
  DANGER = "danger",
}

export enum SIZE {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
}

export enum VARIANT {
  FILL = "fill",
  CLEAR = "clear",
}
