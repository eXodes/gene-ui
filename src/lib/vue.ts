import GBaseButton from "@/components/atoms/base-button/base-button.vue";
import "@/main.css";
import { App } from "vue";

const components = {
  GBaseButton,
};

const install = (Vue: App): void => {
  for (const component in components) {
    // @ts-ignore Expect errors
    Vue.component(components[component].name, components[component]);
  }
};

export default { install };

export { default as GBaseButton } from "@/components/atoms/base-button/base-button.vue";
