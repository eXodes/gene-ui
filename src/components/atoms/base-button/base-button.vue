<template>
  <button
    v-bind="$attrs"
    :class="[baseClass, blockClass, styleClass, sizeClass]"
    :disabled="isDisabled || isLoading"
    class="
      g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50
    "
    data-testid="base-button"
    @click="handleClick"
  >
    <slot>{{ label }}</slot>
  </button>
</template>

<script lang="ts">
  import { defineComponent, computed, PropType } from "vue";
  import { APPEARANCE, SIZE, VARIANT } from "@/constants/attributes";
  import { classNames } from "@/components/atoms/base-button/base-button.classes";

  export default defineComponent({
    name: "GBaseButton",
    props: {
      label: {
        type: String,
        default: null,
      },
      appearance: {
        type: String as PropType<APPEARANCE>,
        default: APPEARANCE.DEFAULT,
        validator: (value: APPEARANCE) => {
          return [
            APPEARANCE.DEFAULT,
            APPEARANCE.PRIMARY,
            APPEARANCE.ATTENTION,
            APPEARANCE.SUCCESS,
            APPEARANCE.WARNING,
            APPEARANCE.DANGER,
          ].includes(value);
        },
      },
      size: {
        type: String as PropType<SIZE>,
        default: SIZE.MEDIUM,
        validator: (value: SIZE) => {
          return [SIZE.SMALL, SIZE.MEDIUM, SIZE.LARGE].includes(value);
        },
      },
      variant: {
        type: String as PropType<VARIANT>,
        default: VARIANT.FILL,
        validator: (value: VARIANT) => {
          return [VARIANT.FILL, VARIANT.CLEAR].includes(value);
        },
      },
      block: {
        type: Boolean,
        default: false,
      },
      isLoading: {
        type: Boolean,
        default: false,
      },
      isDisabled: {
        type: Boolean,
        default: false,
      },
      base: {
        type: Boolean,
        default: false,
      },
    },
    emits: {
      click: null,
    },
    setup(props, { emit }) {
      const isFill = computed(() => props.variant === VARIANT.FILL);

      const baseClass = computed(() => !props.base && "g-rounded-md");
      const blockClass = computed(() => props.block && "g-w-full");

      // eslint-disable-next-line vue/return-in-computed-property
      const sizeClass = computed(() => {
        switch (props.size) {
          case SIZE.SMALL:
            return classNames.size.small;
          case SIZE.MEDIUM:
            return classNames.size.medium;
          case SIZE.LARGE:
            return classNames.size.large;
        }
      });

      // eslint-disable-next-line vue/return-in-computed-property
      const styleClass = computed(() => {
        if (props.isDisabled)
          return isFill.value
            ? classNames.appearance.fill.disabled
            : classNames.appearance.clear.disabled;

        if (props.appearance === APPEARANCE.PRIMARY)
          return isFill.value
            ? classNames.appearance.fill.primary
            : classNames.appearance.clear.primary;

        if (props.appearance === APPEARANCE.ATTENTION)
          return isFill.value
            ? classNames.appearance.fill.attention
            : classNames.appearance.clear.attention;

        if (props.appearance === APPEARANCE.SUCCESS)
          return isFill.value
            ? classNames.appearance.fill.success
            : classNames.appearance.clear.success;

        if (props.appearance === APPEARANCE.WARNING)
          return isFill.value
            ? classNames.appearance.fill.warning
            : classNames.appearance.clear.warning;

        if (props.appearance === APPEARANCE.DANGER)
          return isFill.value
            ? classNames.appearance.fill.danger
            : classNames.appearance.clear.danger;

        if (props.appearance === APPEARANCE.DEFAULT)
          return isFill.value
            ? classNames.appearance.fill.default
            : classNames.appearance.clear.default;
      });

      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const handleClick = (e: any) => emit("click", e);

      return { baseClass, blockClass, styleClass, sizeClass, handleClick };
    },
  });
</script>
