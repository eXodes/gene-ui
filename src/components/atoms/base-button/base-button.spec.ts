import * as stories from "./base-button.stories";
import { composeStories } from "@storybook/testing-vue3";
// TODO: Create an issue related to vite-jest
// import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/vue";
import { APPEARANCE, SIZE, VARIANT } from "@/constants/attributes";
import { classNames } from "@/components/atoms/base-button/base-button.classes";

const { A: Default } = composeStories(stories);

describe("Base Button", () => {
  it("Should render button", async () => {
    render(Default());
    const button = screen.getByTestId("base-button");

    expect(button).toBeVisible();
    expect(button).toMatchSnapshot();
  });

  it("Should render small button", async () => {
    render(
      Default({
        size: SIZE.SMALL,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.size.small);
    expect(button).toMatchSnapshot();
  });

  it("Should render medium button", async () => {
    render(Default());
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.size.medium);
    expect(button).toMatchSnapshot();
  });

  it("Should render large button", async () => {
    render(
      Default({
        size: SIZE.LARGE,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.size.large);
    expect(button).toMatchSnapshot();
  });

  it("Should render default button", async () => {
    render(Default());
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.default);
    expect(button).toMatchSnapshot();
  });

  it("Should render default clear button", async () => {
    render(
      Default({
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.default);
    expect(button).toMatchSnapshot();
  });

  it("Should render primary button", async () => {
    render(
      Default({
        appearance: APPEARANCE.PRIMARY,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.primary);
    expect(button).toMatchSnapshot();
  });

  it("Should render primary clear button", async () => {
    render(
      Default({
        appearance: APPEARANCE.PRIMARY,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.primary);
    expect(button).toMatchSnapshot();
  });

  it("Should render attention button", async () => {
    render(
      Default({
        appearance: APPEARANCE.ATTENTION,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.attention);
    expect(button).toMatchSnapshot();
  });

  it("Should render attention clear button", async () => {
    render(
      Default({
        appearance: APPEARANCE.ATTENTION,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.attention);
    expect(button).toMatchSnapshot();
  });

  it("Should render success button", async () => {
    render(
      Default({
        appearance: APPEARANCE.SUCCESS,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.success);
    expect(button).toMatchSnapshot();
  });

  it("Should render success clear button", async () => {
    render(
      Default({
        appearance: APPEARANCE.SUCCESS,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.success);
    expect(button).toMatchSnapshot();
  });

  it("Should render warning button", async () => {
    render(
      Default({
        appearance: APPEARANCE.WARNING,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.warning);
    expect(button).toMatchSnapshot();
  });

  it("Should render warning clear button", async () => {
    render(
      Default({
        appearance: APPEARANCE.WARNING,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.warning);
    expect(button).toMatchSnapshot();
  });

  it("Should render danger button", async () => {
    render(
      Default({
        appearance: APPEARANCE.DANGER,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.danger);
    expect(button).toMatchSnapshot();
  });

  it("Should render danger clear button", async () => {
    render(
      Default({
        appearance: APPEARANCE.DANGER,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.danger);
    expect(button).toMatchSnapshot();
  });

  it("Should render disabled button", async () => {
    render(
      Default({
        isDisabled: true,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.fill.disabled);
    expect(button).toMatchSnapshot();
  });

  it("Should render disabled clear button", async () => {
    render(
      Default({
        isDisabled: true,
        variant: VARIANT.CLEAR,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass(classNames.appearance.clear.disabled);
    expect(button).toMatchSnapshot();
  });

  it("Should render button as block", async () => {
    render(
      Default({
        block: true,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).toHaveClass("g-w-full");
    expect(button).toMatchSnapshot();
  });

  it("Should render base button", async () => {
    render(
      Default({
        base: true,
      }),
    );
    const button = screen.getByTestId("base-button");

    expect(button).not.toHaveClass("g-rounded-md");
    expect(button).toMatchSnapshot();
  });

  // it("Should be clickable", async () => {
  //   render(Default());
  //   const button = screen.getByTestId("base-button");
  //
  //   userEvent.click(button);
  //   expect(button).toHaveFocus();
  // });

  // it("Should be accessible", async () => {
  //   render(Default());
  //   const button = screen.getByTestId("base-button");
  //
  //   userEvent.tab();
  //   expect(button).toHaveFocus();
  // });
  //
  // it("Should have accessible name", async () => {
  //   render(Default());
  //   const button = screen.getByTestId("base-button");
  //
  //   userEvent.tab();
  //   expect(button).toHaveAccessibleName("Button Text");
  // });
  //
  // it("Should not be clickable when disabled", async () => {
  //   render(
  //     Default({
  //       isDisabled: true,
  //     }),
  //   );
  //   const button = screen.getByTestId("base-button");
  //
  //   userEvent.tab();
  //   expect(button).toBeDisabled();
  // });
  //
  // it("Should not be clickable when loading", async () => {
  //   render(
  //     Default({
  //       isLoading: true,
  //     }),
  //   );
  //   const button = screen.getByTestId("base-button");
  //
  //   userEvent.tab();
  //   expect(button).toBeDisabled();
  // });
});
