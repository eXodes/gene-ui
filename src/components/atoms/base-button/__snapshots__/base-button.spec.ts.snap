// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Button Should render attention button 1`] = `
<button
  class="g-rounded-md g-bg-attention-600 g-text-clear hover:g-bg-attention-500 focus:g-ring-attention-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render attention clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-attention-600 hover:g-bg-attention-100 focus:g-ring-attention-600 dark:hover:g-bg-attention-800 dark:hover:g-bg-opacity-25 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render base button 1`] = `
<button
  class="g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render button 1`] = `
<button
  class="g-rounded-md g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render button as block 1`] = `
<button
  class="g-rounded-md g-w-full g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render danger button 1`] = `
<button
  class="g-rounded-md g-bg-danger-600 g-text-clear hover:g-bg-danger-500 focus:g-ring-danger-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render danger clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-danger-600 hover:g-bg-danger-100 focus:g-ring-danger-600 dark:hover:g-bg-danger-800 dark:hover:g-bg-opacity-25 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render default button 1`] = `
<button
  class="g-rounded-md g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render default clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-gray-600 hover:g-bg-gray-100 focus:g-ring-gray-600 dark:g-text-gray-100 dark:hover:g-bg-gray-800 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render disabled button 1`] = `
<button
  class="g-rounded-md g-bg-gray-200 g-text-gray-400 g-cursor-not-allowed g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
  disabled=""
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render disabled clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-gray-400 g-cursor-not-allowed g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
  disabled=""
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render large button 1`] = `
<button
  class="g-rounded-md g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-2 g-px-4 g-text-md focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render medium button 1`] = `
<button
  class="g-rounded-md g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render primary button 1`] = `
<button
  class="g-rounded-md g-bg-primary-600 g-text-clear hover:g-bg-primary-500 focus:g-ring-primary-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render primary clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-primary-600 hover:g-bg-primary-100 focus:g-ring-primary-600 dark:hover:g-bg-primary-800 dark:hover:g-bg-opacity-25 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render small button 1`] = `
<button
  class="g-rounded-md g-bg-gray-600 g-text-clear hover:g-bg-gray-500 focus:g-ring-gray-600 dark:g-bg-gray-100 dark:g-text-gray-700 dark:hover:g-bg-gray-300 dark:focus:g-ring-gray-600 g-py-1 g-px-2 g-text-xs focus:g-ring-2 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render success button 1`] = `
<button
  class="g-rounded-md g-bg-success-600 g-text-clear hover:g-bg-success-500 focus:g-ring-success-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render success clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-success-600 hover:g-bg-success-100 focus:g-ring-success-600 dark:hover:g-bg-success-800 dark:hover:g-bg-opacity-25 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render warning button 1`] = `
<button
  class="g-rounded-md g-bg-warning-600 g-text-clear hover:g-bg-warning-500 focus:g-ring-warning-600 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;

exports[`Base Button Should render warning clear button 1`] = `
<button
  class="g-rounded-md g-bg-transparent g-text-warning-600 hover:g-bg-warning-100 focus:g-ring-warning-600 dark:hover:g-bg-warning-800 dark:hover:g-bg-opacity-25 g-py-1.5 g-px-3 g-text-sm focus:g-ring-4 g-font-semibold g-tracking-wide g-transition-colors g-duration-100
      focus:g-outline-none focus:g-ring-opacity-50"
  data-testid="base-button"
>
  
  Button Text
  
</button>
`;
